Sources/BSplineLib/ParameterSpaces/parameter_space.hpp

```diff
@@ -156,6 +154,24 @@ RecursiveCombine(const Array<Vector<ValueType>, array_dim>& factors,
                                    result);
 }
 
+template<typename T>
+struct TemporaryArray {
```

Comment:
Did you just reimplement a minimalist version of std::array?

Reply:
I guess it's rather close to std::vector